src/tablib/formats/_xlsx.py

```diff
@@ -166,3 +174,31 @@ def dset_sheet(cls, dataset, ws, freeze_panes=True, escape=False):
 
                 if escape and cell.data_type == 'f' and cell.value.startswith('='):
                     cell.value = cell.value.replace("=", "")
+
+    @classmethod
+    def _adapt_column_width(cls, worksheet, width):
+        if isinstance(width, str) and width != "adaptive":
+            msg = (
+                f"Invalid value for column_width: {width}. "
+                "Must be 'adaptive' or an integer."
+            )
+            raise ValueError(msg)
+
+        if width is None:
+            return
+
+        column_widths = []
+        if width == "adaptive":
+            for row in worksheet.values:
+                for i, cell in enumerate(row):
+                    cell = str(cell)
+                    if len(column_widths) > i:
+                        if len(cell) > column_widths[i]:
+                            column_widths[i] = len(cell)
+                    else:
+                        column_widths.append(len(cell))
```

Comment:
We're not using `cell` for anything else than `len(cell)`,  so perhaps something like this?
```suggestion
                    cell_width = len(str(cell))
                    if len(column_widths) > i:
                        if cell_width > column_widths[i]:
                            column_widths[i] = cell_width
                    else:
                        column_widths.append(cell_width)
```

Reply:
Good catch!